charts/descheduler/templates/cronjob.yaml

```diff
@@ -62,8 +62,6 @@ spec:
             - name: {{ .Chart.Name }}
               image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default (printf "v%s" .Chart.AppVersion) }}"
               imagePullPolicy: {{ .Values.image.pullPolicy }}
-              command:
-                - "/bin/descheduler"
```

Comment:
the cronjob will start and exec the policy-config-file as executable file.
```
OCI runtime create failed: runc create failed: unable to start container process: exec: "--policy-config-file": executable file not found in $PATH: unknown 
```
and you can find the note from k8s doc https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#notes:
```
The command and arguments that you define in the configuration file override the default command and arguments provided by the container image. If you define args, but do not define a command, the default command is used with your new arguments.
```

Reply:
Hey @JaneLiuL ,

does your test run against the currently released version or does it run against the changes I made to the Dockerfile?
What you describe should only happen if the tests run against the current release.

For reference:

This fails, because the current release does not set yet set the ENTRYPOINT in the Dockerfile
```
▶ docker run --rm k8s.gcr.io/descheduler/descheduler:v0.25.0 --policy-config-file /policy-dir/policy.yaml                 
docker: Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "--policy-config-file": executable file not found in $PATH: unknown.
```

This starts the descheduler correctly, because I use the newly built image based on the changes on this branch:
```
▶ docker run --rm 28e98f9bed50 --policy-config-file /policy-dir/policy.yaml
I1028 06:17:25.873278       1 secure_serving.go:210] Serving securely on [::]:10258
I1028 06:17:25.873321       1 tlsconfig.go:240] "Starting DynamicServingCertificateController"
E1028 06:17:25.873397       1 server.go:99] "descheduler server" err="Unable to build in cluster config: unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined"
I1028 06:17:25.873417       1 tlsconfig.go:255] "Shutting down DynamicServingCertificateController"
I1028 06:17:25.873702       1 secure_serving.go:255] Stopped listening on [::]:10258
```

If your tests run against the current release, I could remove the helm and yaml changes until a new release including the Dockerfile change has been rolled out.

Reply:
@JaneLiuL Docker's `ENTRYPOINT` is equivalent to Kubernetes `command` so AFAIK there shouldn't be a change in behavior. Is that correct?